lib/acpi/acpi.c

```diff
@@ -633,17 +633,137 @@ int acpi_table_get(char *signature, int inst, void **acpi_table)
 		status = acpi_early_init();
 		if (status) {
 			LOG_ERR("ACPI early int failed");
-			return status;
+			return NULL;
 		}
 	}
 
 	status = AcpiGetTable(signature, inst, &table);
 	if (ACPI_FAILURE(status)) {
 		LOG_ERR("ACPI get table failed: %d", status);
+		return NULL;
+	}
+
+	return (void *)table;
+}
+
+int acpi_get_madt(int type, struct acpi_subtable_header **tables, int *num_inst, int max_inst)
```

Comment:
actually the function is not even doing what its name means: it is not returning the madt table, but fills in an array of pointer of all (as long as allocation fits, see previous comment), sub-table pointer. 

"acpi_get_madt_entry" would somehow be more semantically right.